test/integration/cert_storage_failed_test.go

```diff
@@ -0,0 +1,198 @@
+//go:build integration
+
+package integration
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/x509"
+	"database/sql"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+	"time"
+
+	_ "github.com/go-sql-driver/mysql"
+	"github.com/letsencrypt/boulder/core"
+	"github.com/letsencrypt/boulder/sa"
+	"github.com/letsencrypt/boulder/test"
+	ocsp_helper "github.com/letsencrypt/boulder/test/ocsp/helper"
+	"golang.org/x/crypto/ocsp"
+)
+
+// getPrecertByName finds and parses a precertificate using the given hostname.
+// It returns the most recent one.
+func getPrecertByName(db *sql.DB, name string) (*x509.Certificate, error) {
+	name = sa.ReverseName(name)
+	// Find the certificate from the precertificates table. We don't know the serial so
+	// we have to look it up by name.
+	var der []byte
+	rows, err := db.Query(`
+		SELECT der
+		FROM issuedNames JOIN precertificates
+		USING (serial)
+		WHERE reversedName = ?
+	`, name)
+	for rows.Next() {
+		err = rows.Scan(&der)
+		if err != nil {
+			return nil, err
+		}
+	}
+	if der == nil {
+		return nil, fmt.Errorf("no precertificate found for %q", name)
+	}
+
+	cert, err := x509.ParseCertificate(der)
+	if err != nil {
+		return nil, err
+	}
+
+	return cert, nil
+}
+
+// expectOCSP500 queries OCSP for the given certificate and expects a 500 error.
+func expectOCSP500(cert *x509.Certificate) error {
+	_, err := ocsp_helper.Req(cert, ocsp_helper.DefaultConfig)
+	if err == nil {
+		return errors.New("Expected error getting OCSP for certificate that failed status storage")
+	}
+
+	var statusCodeError ocsp_helper.StatusCodeError
+	if !errors.As(err, &statusCodeError) {
+		return fmt.Errorf("Got wrong kind of error for OCSP. Expected status code error, got %s", err)
+	} else if statusCodeError.Code != 500 {
+		return fmt.Errorf("Got wrong error status for OCSP. Expected 500, got %d", statusCodeError.Code)
+	}
+	return nil
+}
+
+// TestIssuanceCertStorageFailed tests what happens when a storage RPC fails
+// during issuance. Specifically, it tests that case where we successfully
+// prepared and stored a linting certificate plus metadata, but after
+// issuing the precertificate we failed to mark the certificate as "ready"
+// to serve an OCSP "good" response.
+//
+// To do this, we need to mess with the database, because we want to cause
+// a failure in one specific query, without control ever returning to the
+// client. Fortunately we can do this with MySQL triggers.
+//
+// We also want to make sure we can revoke the precertificate, which we will
+// assume exists (note that this different from the root program assumption
+// that a final certificate exists for any precertificate, though it is
+// similar in spirit).
+func TestIssuanceCertStorageFailed(t *testing.T) {
+	t.Parallel()
+	os.Setenv("DIRECTORY", "http://boulder.service.consul:4001/directory")
+
+	if os.Getenv("BOULDER_CONFIG_DIR") != "test/config-next" {
+		return
```

Comment:
Matt recently showed me that we can call [t.Skip()](https://pkg.go.dev/testing#T.Skip) to skip the test in a more idiomatic way!

Reply:
Very cool. Thanks for sharing the knowledge @aarongable @mcpherrinm.

Reply:
Also worth noting that Copilot proffered a useful and accurate message after t.Skip: `"Skipping test because it requires the StoreLintingCertificateInsteadOfPrecertificate feature flag"`.